### ISSUE TITLE
Focus window to the right / left / below / above not working

### ISSUE THREAD

**Issue by tflori** [domferr/tilingshell#184]
**Describe the bug**
I bound super + wasd to navigate the windows but even though I was able to define them (there where not bound to something else) there are not doing anything

**To Reproduce**
1. Go to Extension Settings
2. Click on View and Customize all the Shortcuts
3. Scroll down to Focus window to the right
4. Click on "Set new Accelerator"
5. Set an unused keybinding
6. Arrange two windows next to each other
7. Activate the window to the left
8. Press the keybinding you set in step 5
9. the window to the right does not get focus

**Screenshots**
These is my keybindings configuration:
![image](https://github.com/user-attachments/assets/273b8aa5-c107-4e74-8b65-a9f7dfcf26b5)

**Information (please complete the following):**
 - Tiling Shell version: 14.1
 - GNOME version: 46

I hope I'm not just miss understanding the functionality of this hotkey.

**Comment by domferr** [comment]
Hey thank you for opening this issue! I tried to reproduce the problem. On my system these hotkeys are already bound to something by default (e.g. SUPER + S opens the overview) and unfortunately I cannot test them. However, the functionality is working correctly if I bind SUPER + WIN + RIGHT / LEFT / UP / DOWN. So, I'm not able to reproduce the problem, I'm afraid. Do you have the same problem if you bind other hotkeys?

**Comment by tflori** [comment]
hm I've installed it on another notebook now and have the same issue - or maybe it is really doing something else?

**Comment by tflori** [comment]
turns out it is a good idea to restart gnome-shell after disabling an extension. back at my office notebook and it works after a fresh boot without changing anything...

**Comment by domferr** [comment]
That's great!

**Comment by tflori** [comment]
oh I have it again. it might have to do with suspending or changing the desktop layout (plug / unplug a monitor).

here is the current output of `journalctl -b|grep tilingshell` https://0x0.st/XdbP.txt

I will try reproducing with a fresh session now.

**Comment by tflori** [comment]
reconnecting the display does not cause the issue but lock/unlock the screen does. the log does not show much difference after logging out and back in:

```
Nov 17 17:35:47 mobile-miracle gnome-shell[5778]: [tilingshell] [extension] building a tiling manager for each monitor
Nov 17 17:35:47 mobile-miracle gnome-shell[5778]: [tilingshell] [TilingManager 0] Work area for monitor 0: 1920 30 3840x1570
Nov 17 17:35:47 mobile-miracle gnome-shell[5778]: [tilingshell] [TilingManager 1] Work area for monitor 1: 0 0 1920x1200
Nov 17 17:35:47 mobile-miracle gnome-shell[5778]: [tilingshell] [extension] extension is enabled
Nov 17 17:35:48 mobile-miracle gnome-shell[5778]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
Nov 17 17:36:02 mobile-miracle gnome-shell[145366]: Extension tilingshell@ferrarodomenico.com already installed in /usr/share/gnome-shell/extensions/tilingshell@ferrarodomenico.com. /usr/share/gnome-shell/extensions/tilingshell@ferrarodomenico.com will not be loaded
Nov 17 17:36:15 mobile-miracle gnome-shell[146009]: [tilingshell] [extension] extension is enabled
Nov 17 17:36:15 mobile-miracle gnome-shell[146009]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
Nov 17 17:36:15 mobile-miracle gnome-shell[146009]: [tilingshell] [extension] building a tiling manager for each monitor
Nov 17 17:36:15 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 0] Work area for monitor 0: 1920 0 3840x1600
Nov 17 17:36:15 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 1] Work area for monitor 1: 0 0 1920x1200
Nov 17 17:36:15 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 0] new work area for monitor 0: 1920 30 3840x1570
Nov 17 17:37:49 mobile-miracle gnome-shell[146009]: [tilingshell] [LayoutWidget] relayout not needed
Nov 17 17:37:49 mobile-miracle gnome-shell[146009]: [tilingshell] [LayoutWidget] relayout not needed
Nov 17 17:37:49 mobile-miracle gnome-shell[146009]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
Nov 17 17:38:21 mobile-miracle gnome-shell[146009]: [tilingshell] [LayoutWidget] relayout not needed
Nov 17 17:38:21 mobile-miracle gnome-shell[146009]: [tilingshell] [LayoutWidget] relayout not needed
Nov 17 17:38:21 mobile-miracle gnome-shell[146009]: [tilingshell] [extension] building a tiling manager for each monitor
Nov 17 17:38:21 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 0] Work area for monitor 0: 0 0 1920x1200
Nov 17 17:38:21 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 0] new work area for monitor 0: 0 30 1920x1170
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [LayoutWidget] relayout not needed
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [extension] building a tiling manager for each monitor
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 0] Work area for monitor 0: 1920 0 3840x1600
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 1] Work area for monitor 1: 0 30 1920x1170
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 0] new work area for monitor 0: 1920 30 3840x1570
Nov 17 17:38:53 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 1] new work area for monitor 1: 0 0 1920x1200
Nov 17 17:39:18 mobile-miracle gnome-shell[146009]: [tilingshell] [LayoutWidget] relayout not needed
Nov 17 17:39:34 mobile-miracle gnome-shell[146009]: [tilingshell] [extension] extension is disabled
Nov 17 17:39:51 mobile-miracle gnome-shell[146009]: [tilingshell] [extension] building a tiling manager for each monitor
Nov 17 17:39:51 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 0] Work area for monitor 0: 1920 30 3840x1570
Nov 17 17:39:51 mobile-miracle gnome-shell[146009]: [tilingshell] [TilingManager 1] Work area for monitor 1: 0 0 1920x1200
Nov 17 17:39:51 mobile-miracle gnome-shell[146009]: [tilingshell] [extension] extension is enabled
Nov 17 17:39:52 mobile-miracle gnome-shell[146009]: [tilingshell] [DefaultMenu] [{"name":null,"x":1920,"y":0,"width":3840,"height":1600},{"name":null,"x":0,"y":0,"width":1920,"height":1200}]
 iras  ~  

```

this log includes all tests (unplugging the monitor too) and to disable / re-enable the extension didn't work either.

**Comment by tflori** [comment]
I just verified that this also happens when I disable all other extensions.

**Comment by jonestristand** [comment]
I am also having this issue, even with other extensions disabled. No action happens at all when using the assigned focus accelerators.

**Comment by domferr** [comment]
Hey guys, quick update for you! It turns out there is a bug in Tiling Shell, isn't it? :smile: 

The good news is that a bugfix is ready and will be shipped with the next update soon :soon:

**Comment by domferr** [comment]
Hey, quick update. Meanwhile the bugfix gets published, I'll leave the update to you in advance!

- For GNOME Shell 42, 43 or 44: [GNOME.42-44.tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/17922215/GNOME.42-44.tilingshell%40ferrarodomenico.com.zip)
- For GNOME Shell 45, 46 or 47: [tilingshell@ferrarodomenico.com.zip](https://github.com/user-attachments/files/17922218/tilingshell%40ferrarodomenico.com.zip)

**Comment by lucasj96** [comment]
Had the same issue, restarted my desktop and it started working.
Maybe it's helfpful but had previously installed pop-shell for fedora prior to installing tilingshell without restarting gnome.

**Comment by domferr** [comment]
:heavy_check_mark: Available in version 15.1! The new version is already available on GSE
<div align="center">
  <a href="https://extensions.gnome.org/extension/7065/tiling-shell/" >
      <img src="https://img.shields.io/badge/Install%20from-extensions.gnome.org-4A86CF?style=for-the-badge&logo=Gnome&logoColor=white"/>
  </a>
</div>
